File: lua/day1_part1_2025.lua

local f = assert(io.open("input.txt", "r"))
local pos, cnt = 50, 0
for token in f:read("*a"):gmatch("%S+") do
    local dir, amt = token:sub(1,1), tonumber(token:sub(2))
    pos = (pos + (dir == "R" and amt or -amt)) % 100
    cnt = cnt + (pos == 0 and 1 or 0)
end
f:close()
print(cnt)

File: r/day1_part1_2025.R

input <- scan("input.txt", what = character(), quiet = TRUE)
pos <- 50
cnt <- 0

for (s in input) {
  dir <- substr(s, 1, 1)
  amt <- as.integer(substr(s, 2, nchar(s)))
  pos <- (pos + if (dir == "R") amt else -amt) %% 100
  if (pos == 0) cnt <- cnt + 1
}

cat(cnt, "\n")
